### ISSUE TITLE
Will there be a nested selector inheritance?

### ISSUE THREAD

**Issue by firedev** [sass/sass#609]
I understand that nested selectors are not supported in @extend. But will they be? Like in Stylus.

```
.description .price {}
.otherplace .price {
    @extend .description .price;
}
```

Should compile to:

```
.description .price, .otherplace .price {}
```

Is there any reason why it is not supported? What is the best practice around this limitation? Mixins?

**Comment by Snugug** [comment]
I think what you're looking for is a placeholder selector w/extend.

``` scss
%price {}

.description .price {
  @extend %price;
}

.otherplace .price {
  @extend %price;
}
```

**Comment by firedev** [comment]
That's what I thought, thanks.

But what is the official stance on this? Will there be a nested selector inheritance in the future or not?

**Comment by Snugug** [comment]
IIRC placeholder selectors were the official response to this requested functionality.

**Comment by nex3** [comment]
Sass doesn't support extending nested selectors because the way it would need to work to be consistent with the overall semantics of `@extend` would be unintuitive for a lot of users.

To understand why this is, you need to understand the semantics of `@extend`. When you write `A { @extend B }`, what you're saying is that all elements matching `A` should be styled as though they match `B` as well. So when you have `.foo { @extend .bar .baz }`, what that means is that all elements matching `.foo` should be styled as though they also match `.bar .baz`.

Now, if an element matches `.bar .baz`, that means it also matches `.baz` on its own. So you get the following behavior:

``` scss
.foo { @extend .bar .baz }
.bar .baz { a: b }
.baz { b: c }
```

which would compile to:

``` css
.bar .baz, .foo { a: b }
.baz, .foo { b: c }
```

That's all well and good. It's not widely useful, since it overlaps almost entirely with `@extend .baz`, but we'd support it if there weren't two easy misinterpretations of the semantics that would confuse users.

### Misinterpretation 1: Search-and-Replace

A lot of users think of `@extend` as a sort of "search-and-replace" for selectors. It's actually much more powerful than that, but that's a pretty good rule of thumb for simple cases. However, that rule of thumb breaks down for nested selectors, since `.foo { @extend .bar .baz }` also implies `.foo { @extend .baz }`. This can cause unexpected replacements of `.baz` on its own with `.foo`, which is a problem.

Incidentally, this is the misinterpretation to which Stylus subscribes. For example, in Stylus:

``` stylus
.bar .baz { c: d }
.baz { e: f }
.foo {
  @extend .bar .baz;
}
```

compiles to

``` css
.bar .baz,
.foo {
  c: d;
}
.baz {
  e: f;
}
```

### Misinterpretation 2: Namespaced Inheritance

Another way to misinterpret `.foo { @extend .bar .baz }` is to consider the nesting to mean something similar to what it does in the context of a CSS rule: `.foo` should be styled as though it matches `.baz` but only within the context of `.bar`. This would mean that

``` scss
.foo { @extend .bar .baz }
.bar .baz { a: b }
.baz { b: c }
```

would compile to

``` css
.bar .baz, .bar .foo { a: b }
.baz { b: c }
```

which is wildly different than the output generated by the real semantics of `@extend`.

### tl;dr

There's no way to support nested inheritance that would make sense to all users, and the way that best fits the semantics of `@extend` isn't especially useful.
